unit_testing/test_reports.py

```diff
@@ -9,30 +9,57 @@
 from spinalcordtoolbox.reports.slice import Sagittal
 
 
-def labeled_data_test_params(path_in='sct_testing_data/t2/t2.nii.gz',
-                             path_seg='sct_testing_data/t2/labels.nii.gz'):
-    """Generate image/label pairs for various test cases of
-    test_sagittal_slice_get_center_spit."""
-    im_in = Image(path_in)            # Base anatomical image
-    im_seg_labeled = Image(path_seg)  # Base labeled segmentation
-    assert np.count_nonzero(im_seg_labeled.data) >= 2, "Labeled segmentation image has fewer than 2 labels"
+@pytest.fixture(scope="module")
+def im_base(path_in=sct_test_path('t2', 't2.nii.gz')):
+    # Base anatomical image
+    yield Image(path_in)
+
+
+@pytest.fixture(scope="module")
+def im_seg_labeled(path_seg=sct_test_path('t2', 'labels.nii.gz')):
+    # Base labeled segmentation
+    im_seg = Image(path_seg)
+    assert np.count_nonzero(im_seg.data) >= 2, "Labeled segmentation image has fewer than 2 labels"
+    yield im_seg
 
+
+@pytest.fixture(scope="module")
+def im_seg_one_label(im_seg_labeled):
     # Create image with all but one label removed
     im_seg_one_label = im_seg_labeled.copy()
     for x, y, z in np.argwhere(im_seg_one_label.data)[1:]:
         im_seg_one_label.data[x, y, z] = 0
+    yield im_seg_one_label
+
 
+@pytest.fixture(scope="module")
+def im_seg_no_labels(im_seg_labeled):
     # Create image with no labels
     im_seg_no_labels = im_seg_labeled.copy()
     for x, y, z in np.argwhere(im_seg_no_labels.data):
         im_seg_no_labels.data[x, y, z] = 0
+    yield im_seg_no_labels
+
+
+def labeled_data_test_params():
+    """Generate image/label pairs for various test cases of
+    test_sagittal_slice_get_center_spit."""
+    return [pytest.param('im_base', 'im_seg_labeled', id='multiple_labels'),
+            pytest.param('im_base', 'im_seg_one_label', id='one_label'),
+            pytest.param('im_base', 'im_seg_no_labels', id='no_labels')]
+
+
+@pytest.fixture
+def im_in(request):
+    return request.getfixturevalue(request.param)
```

Comment:
I'm having second thoughts about this workaround... 

I came across [this very well-written SO comment](https://stackoverflow.com/a/56865893) that explains why Pytest isn't designed to mix fixtures and parameterization.

The key point: Fixtures are designed to be run after the tests have already been collected. So, I'm convinced that the tests (and their data) need to be rewritten in one of two ways:
1. Create the data for the tests ahead of time, so that it's already available before tests are collected.
   * Cons: Pytest won't know what tests are being run, so test data will have to be prepared every time, even if it's not needed.
2. Stop using `pytest.mark.parameterize` for multiple input types. Instead, write 1 test per kind of input, so that pytest can collect the tests independent of any fixtures being used. 

I'm leaning towards 2. In fact, that's how I wrote tests when I was contributing to `scikit-learn`. Over there, each test corresponds to 1 type of input. [Here's a test](https://github.com/scikit-learn/scikit-learn/blob/2218ec46227c92301ac6837c4a8ae9b8dc5d3960/sklearn/manifold/tests/test_t_sne.py#L895-L923) I wrote over there; in this case, the test is named for the type of input: "`test_tsne_different_square_distances`". 

`pytest.mark.parameterize` stills gets used. But, the things that _do_ get parameterized are other, unrelated variables, not the key inputs that trigger certain behaviors.

Reply:
Specifically, I think 1) is appropriate for `test_labels.py` (since the same dummy images get used across multiple tests), while 2) is appropriate for `test_reports.py` (where the input data is specific to the a single test, so just break the test up).

Reply:
Ah, it looks like there might be a better way: [`pytest-cases`](https://smarie.github.io/python-pytest-cases/#why-pytest-cases). I'm going to try experimenting with this plugin, as it seems very well-established and maintained.